File: c#/src/Docmost.Sdk/Services/UsersApi.cs
using Docmost.Sdk.Http;
using Docmost.Sdk.Models;
using System.Net.Http.Headers;
using System.Net.Http.Json;
using System.Text.Json;

namespace Docmost.Sdk.Services;

public sealed class UsersApi(DocmostHttpClient http)
{
    private readonly DocmostHttpClient _http = http;

    public async Task<ApiResponse<JsonElement?>> GetUserInfoAsync(CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("users/me", null, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> UpdateUserAsync(UpdateUserDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("users/update", request, cancellationToken).ConfigureAwait(false);
    }

}

File: typescript/src/services/users-api.ts
/** Generated API client. */
import type { ApiResponse } from '../types.js';
import type { DocmostHttpClient } from '../http-client.js';
import type { UpdateUserDto } from '../models/index.js';


export class UsersApi {
  constructor(private readonly http: DocmostHttpClient) {}

  async getUserInfo(): Promise<ApiResponse<unknown>> {
    return this.http.post(`users/me`, undefined);
  }

  async updateUser(request: UpdateUserDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`users/update`, request);
  }

}
